
# <i>Okay, so you're Brad Pitt:</i><br><br><center>An Analysis of Movie Revenues based on Actors and Movie Genres</center>


## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#wrangling">Data Wrangling</a></li>
<li><a href="#eda">Exploratory Data Analysis</a></li>
<li><a href="#conclusions">Conclusions</a></li>
</ul>

<a id='intro'></a>
## Introduction

As English-American actor Bob Hope put it,

> <i>Celebrities have a way of touching our lives. Perhaps we are influenced by their screen image, or perhaps by their acquired status [...]</i> 

<p style='text-align: justify'>
Celebrities have a certain charm, some more than others, which makes us want to watch them. In the context of movies, this can - but not necessarily - translate to higher revenues for films and the companies that produced them. This "charm" aspect of actors and its effect on the marketability and financial success of movies has been the topic in academic literature for many years.

The empirical analysis undertaken in this project aims to contribute to this discussion by examining the TMDB dataset containing data on movies from 1960 to 2017. In particular, this analysis aims to find out if a correlation exists between (top) stars and movie revenues. Another goal of this project is to show (or attempt to show) if an interplay between the actors, genres, and the invested budget correlates with the commercial success of movies.
</p>
<br>


In [1]:
# Import packages for the analysis
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

<a id='wrangling'></a>
## Data Wrangling

To start the analysis, the TMDB dataset needs to be loaded in a Pandas dataframe on which cleaning methods will be applied.

### General Properties

In [2]:
# load the dataset in a dataframe and show the first five rows

main_df = pd.read_csv(r"C:\Users\49176\Desktop\Udacity\Data Analyst Nanodegree\Project 2 - Investigate a Dataset\tmdb-movies.csv", header = 0, index_col = 'id' )
main_df.head()

imdb_id  popularity     budget     revenue  \
id                                                     
135397  tt0369610   32.985763  150000000  1513528810   
76341   tt1392190   28.419936  150000000   378436354   
262500  tt2908446   13.112507  110000000   295238201   
140607  tt2488496   11.173104  200000000  2068178225   
168259  tt2820852    9.335014  190000000  1506249360   

                      original_title  \
id                                     
135397                Jurassic World   
76341             Mad Max: Fury Road   
262500                     Insurgent   
140607  Star Wars: The Force Awakens   
168259                     Furious 7   

                                                     cast  \
id                                                          
135397  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
76341   Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
262500  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
140607  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
168259  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   

                                                 homepage          director  \
id                                                                            
135397                      http://www.jurassicworld.com/   Colin Trevorrow   
76341                         http://www.madmaxmovie.com/     George Miller   
262500     http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   
140607  http://www.starwars.com/films/star-wars-episod...       J.J. Abrams   
168259                           http://www.furious7.com/         James Wan   

                              tagline  \
id                                      
135397              The park is open.   
76341              What a Lovely Day.   
262500     One Choice Can Destroy You   
140607  Every generation has a story.   
168259            Vengeance Hits Home   

                                                 keywords  \
id                                                          
135397  monster|dna|tyrannosaurus rex|velociraptor|island   
76341    future|chase|post-apocalyptic|dystopia|australia   
262500  based on novel|revolution|dystopia|sequel|dyst...   
140607              android|spaceship|jedi|space opera|3d   
168259                car race|speed|revenge|suspense|car   

                                                 overview  runtime  \
id                                                                   
135397  Twenty-two years after the events of Jurassic ...      124   
76341   An apocalyptic story set in the furthest reach...      120   
262500  Beatrice Prior must confront her inner demons ...      119   
140607  Thirty years after defeating the Galactic Empi...      136   
168259  Deckard Shaw seeks revenge against Dominic Tor...      137   

                                           genres  \
id                                                  
135397  Action|Adventure|Science Fiction|Thriller   
76341   Action|Adventure|Science Fiction|Thriller   
262500         Adventure|Science Fiction|Thriller   
140607   Action|Adventure|Science Fiction|Fantasy   
168259                      Action|Crime|Thriller   

                                     production_companies release_date  \
id                                                                       
135397  Universal Studios|Amblin Entertainment|Legenda...       6/9/15   
76341   Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15   
262500  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15   
140607          Lucasfilm|Truenorth Productions|Bad Robot     12/15/15   
168259  Universal Pictures|Original Film|Media Rights ...       4/1/15   

        vote_count  vote_average  release_year    budget_adj   revenue_adj  
id                                                                          
135397        5562           6.5          2015  1.379999e+08  1.392446e+09  
76341         6185       

In [3]:
# Show the number of rows and columns in the dataframa
main_df.shape

(10866, 20)

The dataset contains **10866** rows with **20** columns. The dataset was loaded with the column $\mathit{id}$ as the index.

In [4]:
# Show the data types of each column
main_df.dtypes

imdb_id                  object
popularity              float64
budget                    int64
revenue                   int64
original_title           object
cast                     object
homepage                 object
director                 object
tagline                  object
keywords                 object
overview                 object
runtime                   int64
genres                   object
production_companies     object
release_date             object
vote_count                int64
vote_average            float64
release_year              int64
budget_adj              float64
revenue_adj             float64
dtype: object

We see here that the there are a lot of columns (11, to be exact) which are of type $\mathit{object}$. These are mostly text (string) fields about the movies in our dataset. It is important to know these fields as data manipulation largely relies on the data type.

In [5]:
# Show string fields of the dataframe
main_df.select_dtypes(include = 'object').head()

imdb_id                original_title  \
id                                                
135397  tt0369610                Jurassic World   
76341   tt1392190            Mad Max: Fury Road   
262500  tt2908446                     Insurgent   
140607  tt2488496  Star Wars: The Force Awakens   
168259  tt2820852                     Furious 7   

                                                     cast  \
id                                                          
135397  Chris Pratt|Bryce Dallas Howard|Irrfan Khan|Vi...   
76341   Tom Hardy|Charlize Theron|Hugh Keays-Byrne|Nic...   
262500  Shailene Woodley|Theo James|Kate Winslet|Ansel...   
140607  Harrison Ford|Mark Hamill|Carrie Fisher|Adam D...   
168259  Vin Diesel|Paul Walker|Jason Statham|Michelle ...   

                                                 homepage          director  \
id                                                                            
135397                      http://www.jurassicworld.com/   Colin Trevorrow   
76341                         http://www.madmaxmovie.com/     George Miller   
262500     http://www.thedivergentseries.movie/#insurgent  Robert Schwentke   
140607  http://www.starwars.com/films/star-wars-episod...       J.J. Abrams   
168259                           http://www.furious7.com/         James Wan   

                              tagline  \
id                                      
135397              The park is open.   
76341              What a Lovely Day.   
262500     One Choice Can Destroy You   
140607  Every generation has a story.   
168259            Vengeance Hits Home   

                                                 keywords  \
id                                                          
135397  monster|dna|tyrannosaurus rex|velociraptor|island   
76341    future|chase|post-apocalyptic|dystopia|australia   
262500  based on novel|revolution|dystopia|sequel|dyst...   
140607              android|spaceship|jedi|space opera|3d   
168259                car race|speed|revenge|suspense|car   

                                                 overview  \
id                                                          
135397  Twenty-two years after the events of Jurassic ...   
76341   An apocalyptic story set in the furthest reach...   
262500  Beatrice Prior must confront her inner demons ...   
140607  Thirty years after defeating the Galactic Empi...   
168259  Deckard Shaw seeks revenge against Dominic Tor...   

                                           genres  \
id                                                  
135397  Action|Adventure|Science Fiction|Thriller   
76341   Action|Adventure|Science Fiction|Thriller   
262500         Adventure|Science Fiction|Thriller   
140607   Action|Adventure|Science Fiction|Fantasy   
168259                      Action|Crime|Thriller   

                                     production_companies release_date  
id                                                                      
135397  Universal Studios|Amblin Entertainment|Legenda...       6/9/15  
76341   Village Roadshow Pictures|Kennedy Miller Produ...      5/13/15  
262500  Summit Entertainment|Mandeville Films|Red Wago...      3/18/15  
140607          Lucasfilm|Truenorth Productions|Bad Robot     12/15/15  
168259  Universal Pictures|Original Film|Media Rights ...       4/1/15

As stated in this <a href='https://s3.amazonaws.com/video.udacity-data.com/topher/2018/July/5b57919a_data-set-options/data-set-options.pdf'>document</a>, some columns contain multiple values which are separated by delimiters. This can be observed in the columnds *cast*, *genres*, *production_companies*, and *keywords*. As the focus of this analysis is on actors and genres, the values in these columns need to be separated.
<br><br>
It can also be observed that the dataframe has two id columns: *id* and *imdb_id*. The former (*id*) was used as an index column when the dataset was loaded as a dataframe. It does not make sense here to have two id columns. Dropping the *imdb_id* column would be an obvious solution here.
Furthermore &mdash; at least for the purposes of this analysis &mdash; the columns *homepage* , *overview*, *tagline*, and *keywords* are not necessary. A movie with a website could potentially attract more movie goers, especially in this social media age, and a catchy tagline sticks. Both of these could translate to higher revenues but will not be the focus of this analysis.

In [6]:
# show summary statistics for numerical columns
main_df.describe()

popularity        budget       revenue       runtime    vote_count  \
count  10866.000000  1.086600e+04  1.086600e+04  10866.000000  10866.000000   
mean       0.646441  1.462570e+07  3.982332e+07    102.070863    217.389748   
std        1.000185  3.091321e+07  1.170035e+08     31.381405    575.619058   
min        0.000065  0.000000e+00  0.000000e+00      0.000000     10.000000   
25%        0.207583  0.000000e+00  0.000000e+00     90.000000     17.000000   
50%        0.383856  0.000000e+00  0.000000e+00     99.000000     38.000000   
75%        0.713817  1.500000e+07  2.400000e+07    111.000000    145.750000   
max       32.985763  4.250000e+08  2.781506e+09    900.000000   9767.000000   

       vote_average  release_year    budget_adj   revenue_adj  
count  10866.000000  10866.000000  1.086600e+04  1.086600e+04  
mean       5.974922   2001.322658  1.755104e+07  5.136436e+07  
std        0.935142     12.812941  3.430616e+07  1.446325e+08  
min        1.500000   1960.000000  0.000000e+00  0.000000e+00  
25%        5.400000   1995.000000  0.000000e+00  0.000000e+00  
50%        6.000000   2006.000000  0.000000e+00  0.000000e+00  
75%        6.600000   2011.000000  2.085325e+07  3.369710e+07  
max        9.200000   2015.000000  4.250000e+08  2.827124e+09

As already mentioned, the data cleaning techniques vary based on the data type. We can see in these summary statistics that the minimum value in the *budget* and *revenue* as well as in the adjusted *budget_adj* and *revenue_adj* columns is 0. Furthermore, the adjusted columns are of type $\textit{float64}$ as shown above. The *budget* and *bevenue* columns (and the columns with the adjusted values) have a minimum of zero which looks very anomalous. Movies cannot have zero budgets nor zero revenue and these missing values accordingly need to be imputed. The *budget* and *revenue* will obviously have direct positive correlations with the *budjet_adj* and *revenue_adj* columns since the adjusted value columns were created from the former. For ease of comparison, only the adjusted columns will be used in this analysis.


### Data Cleaning

For data cleaning purposes in this analysis, the following have to be done:

<ul>
    <li>Remove the <i>imdb</i> id since the column <i>id</i> has already been used as index.</li>
    <li>Split the values in the <i>actors</i> and <i>genres</i> into different columns.</li>
    <li>Remove the columns <i>homepage, overview, tagline, director</i> and <i>keywords</i> as extraneous string fields.</li>
    <li>Remove the superfluous <i>budget</i> and <i>revenue</i> columns and retain the adjusted values (<i>budget_adj</i> and <i>revenue_adj</i>).</li>
    <li>Impute the zero values in the <i>budget_adj</i> and <i>revenue_adj</i>.
</ul>

In [ ]:
# After discussing the structure of the data and any problems that need to be
#   cleaned, perform those cleaning steps in the second part of this section.


<a id='eda'></a>
## Exploratory Data Analysis

### Research Question 1 (Replace this header name!)

In [ ]:
# Use this, and more code cells, to explore your data. Don't forget to add
#   Markdown cells to document your observations and findings.


### Research Question 2  (Replace this header name!)

In [ ]:
# Continue to explore the data to address your additional research
#   questions. Add more headers as needed if you have more questions to
#   investigate.


<a id='conclusions'></a>
## Conclusions

